packages/forms-web-app/src/controllers/examination/file-upload/fileValidation.js

```diff
@@ -0,0 +1,107 @@
+const { calcMaxFileSizeLimit } = require('./utils');
+
+const allowedMimeTypes = [
+	'application/msword',
+	'application/pdf',
+	'application/vnd.ms-excel',
+	'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+	'image/bmp',
+	'image/jpeg',
+	'image/png',
+	'image/tiff'
+];
+
+const {
+	routesConfig: {
+		examination: {
+			pages: { selectFile }
+		}
+	}
+} = require('../../../routes/config');
+
+const {
+	fileUpload: { amountOfFileslimit, maxFileSizeInMb }
+} = require('../../../config');
+const { saveFileToDisk } = require('./fileManagement');
+const { addFileToSession } = require('./fileSessionManagement');
+
+const {
+	maxSize: maxSizeMessage,
+	checkMimeType: checkMimeTypeMessage,
+	isEmpty: isEmptyMessage
+} = selectFile.onError.message;
+
+const { noFileSelected: noFileSelectedMessage, moreThanXFiles: moreThanXAmountFilesMessage } =
+	selectFile.onError.message;
+
+const moreThanXAmountFiles = (fileToUpload, filesInSession) => {
+	const errorSummary = [];
+	let errorMessage = '';
+	const totalItems = filesInSession.length + fileToUpload.length;
+	if (totalItems > amountOfFileslimit || fileToUpload.length > amountOfFileslimit) {
+		errorMessage = { text: moreThanXAmountFilesMessage(amountOfFileslimit) };
+		errorSummary.push({ text: moreThanXAmountFilesMessage(amountOfFileslimit) });
+	}
+
+	return {
+		errorMessage,
+		errorSummary
+	};
+};
+
+const noFileSelected = (files) => {
+	const errorSummary = [];
+	let errorMessage = '';
+	if (!files || Object.keys(files).length === 0) {
+		errorMessage = { text: noFileSelectedMessage };
+		errorSummary.push({ text: noFileSelectedMessage });
+	}
+
+	return {
+		errorMessage,
+		errorSummary
+	};
+};
+
+const validationHandler = (fileToUpload) => {
+	const {
+		mimetype: fileToUploadMimeType,
+		size: fileToUploadSize,
+		name: fileToUploadName
+	} = fileToUpload;
+
+	if (fileToUploadSize === 0) return isEmptyMessage(fileToUploadName);
+	if (fileToUploadSize > calcMaxFileSizeLimit())
+		return maxSizeMessage(fileToUploadName, maxFileSizeInMb);
+	if (!allowedMimeTypes.includes(fileToUploadMimeType))
+		return checkMimeTypeMessage(fileToUploadName);
+};
+
+const handleMultipleFileUploadsWithErrors = async (session, fileToUpload) => {
+	const errorSummary = [];
+	let errorMessage = '';
+	for (const document of fileToUpload) {
+		const error = validationHandler(document);
+		if (!error) {
+			const savedFileData = await saveFileToDisk(document);
```

Comment:
this feels necessary to save as we go rather than get the good and bad array and loop again ?